File: backend/models/Nutrition.js
import mongoose from 'mongoose';

const NutritionSchema = new mongoose.Schema({
    client: { type: mongoose.Schema.Types.ObjectId, ref: 'Client', required: true, unique: true },
    meals: [
        {
            mealName: { type: String, required: true },
            items: [
                {
                    name: { type: String, required: true },
                    quantity: { type: String, required: true }
                }
            ]
        }
    ]
});

export default mongoose.model('Nutrition', NutritionSchema);

File: MyPTbook/Models/Nutrition.swift
import Foundation

struct Nutrition: Codable, Identifiable {
    let _id: String
    let client: String
    var meals: [Meal]
    
    var id: String { _id }
    
    private enum CodingKeys: String, CodingKey {
        case _id, client, meals
    }
    
    struct Meal: Codable, Identifiable {
        let mealName: String
        var items: [MealItem]
        
        var id: String { mealName }
        
        private enum CodingKeys: String, CodingKey {
            case mealName, items
        }
        
        init(mealName: String, items: [MealItem] = []) {
            self.mealName = mealName
            self.items = items
        }
    }
    
    struct MealItem: Codable, Identifiable {
        let name: String
        let quantity: String
        
        var id: String { name }
        
        private enum CodingKeys: String, CodingKey {
            case name, quantity
        }
        
        init(name: String, quantity: String) {
            self.name = name
            self.quantity = quantity
        }
    }
} 